compiler/rustc_codegen_llvm/src/intrinsic.rs

```diff
@@ -160,6 +160,14 @@ impl<'ll, 'tcx> IntrinsicCallMethods<'tcx> for Builder<'_, 'll, 'tcx> {
                     _ => bug!("the va_arg intrinsic does not work with non-scalar types"),
                 }
             }
+            // FIXME: why is this needed when `type_id` is not?
```

Comment:
I'm curious to know what fails when this is missing?

Reply:
`error: internal compiler error: compiler/rustc_codegen_llvm/src/intrinsic.rs:381:18: unknown intrinsic 'type_id128' -- should it have been lowered earlier?`

Reply:
Sorry, I mean when exactly it's failing. Stage 0 library build, stage 1 library build (core, std, later?), or a specific ui test?

Also, the whole stack trace may be useful too.

Reply:
Oh, during `stage1` build of library (specifically `alloc`): https://gist.github.com/thomcc/1408507eab7b6e0b82ab97670236663a

Reply:
Huh -- then @scottmcm is right. We should be handling this at compiler/rustc_codegen_ssa/src/mir/intrinsic.rs:130...

Reply:
Hm, it might not be needed. Things seem to pass locally without it now, I wonder if I had forgotten to save a file somewhere?

Reply:
I just pulled down the changes locally and it built+tested just fine for me too.